mono/utils/mono-threads.c

```diff
@@ -512,7 +512,7 @@ unregister_thread (void *arg)
 	MonoThreadHandle* handle;
 
 	info = (MonoThreadInfo *) arg;
-	g_assert (info);
+	g_assert (info && "unregister_thread");
 	g_assert (mono_thread_info_is_current (info));
```

Comment:
The assert already contains a line number which can be used to identify the location if the exact version of the runtime is known.

Reply:
Agreed, but I didn't see version in the issue. Still could lean on users to report that, or derive it from "Unity Xyz."

Reply:
@jaykrell you yourself added `g_assertf`, why not using it? statement vs expression problem?

Reply:
I think either is ok. ‘f’ is required for dynamic values. Non-f suffices for constants.

Reply:
Maybe ‘&&’ is obscure, making reader evaluate the true/falseness of it? ‘f’ nicely separates expression checked from reported information?

Reply:
I had forgotten that `g_assertf` already includes `__func__`.
Perhaps it should honor `DISABLE_ASSERT_MESSAGES`.